lib/internal/wrapAsync.js

```diff
@@ -0,0 +1,27 @@
+import identity from 'lodash/identity';
+import asyncify from '../asyncify';
+
+var supportsSymbol = typeof Symbol === 'function';
+
+function supportsAsync() {
+    var supported;
+    try {
+        /* eslint no-eval: 0 */
+        supported = isAsync(eval('(async function () {})'));
```

Comment:
`new Function()` and `eval` are equivalently evil.  Here, they are necessary, because there is no other way to detect `async` support without causing a parse error that halts JS execution.